src/worker.js

```diff
@@ -169,9 +175,11 @@ worker.on('message', function (request) {
               currentRequestId = null;
             })
             .catch(function (err) {
+              controller && controller.abort(convertError(err));
               worker.send({
                 id: request.id,
                 result: null,
+                aborted: controller ? true : false,
```

Comment:
The boolean is for the case of "offloaded functions" since they might result in an `error !== null` but do not have a controller associated with the execution. So this boolean allows us to be sure that the event handler should be invoked.